src/transaction.rs

```diff
@@ -176,16 +178,30 @@ impl Transaction {
     }
 
     pub fn input_colors(&self) -> impl Iterator<Item = &Color> {
```

Comment:
Why break this API and make a new one to do the same thing?

Reply:
This wasn't changed: https://github.com/FuelLabs/fuel-tx/blob/d37cb3b0cdc02ed120f7ccbe5648fb8c8205a228/src/transaction.rs#L178

Reply:
This API used to dedupe the colors, but now the dedupe logic was moved to a separate api.